library/src/main/java/in/krharsh17/barview/BarView.java

```diff
@@ -4,26 +4,50 @@
 import android.content.res.TypedArray;
 import android.util.AttributeSet;
 import android.util.Log;
+import android.view.GestureDetector;
+import android.view.MotionEvent;
+import android.view.View;
 import android.view.ViewGroup;
 import android.widget.LinearLayout;
 import android.widget.ScrollView;
 
-
 import androidx.constraintlayout.widget.ConstraintLayout;
 
 import java.util.ArrayList;
 import java.util.List;
 
 public class BarView extends ScrollView implements Constants {
+
     LinearLayout containerLayout;
     Context context;
-
+    private GestureDetector gestureDetector;
+    private onbarClickListener onBarClickListener;
     List<BarGroup> barGroups;
     List<BarModel> data;
 
     private int BAR_MARGIN = 6, VERTICAL_SPACING = 48, BAR_HEIGHT = 20, LABEL_FONT_SIZE = 18, VALUE_FONT_SIZE = 9;
     private String LABEL_TEXT_COLOR = labelTextColor, VALUE_TEXT_COLOR = valueTextColor, RIPPLE_COLOUR = rippleColor;// has to be >2
 
+
+    /**
+     * Returns a reference to the attached Listener
+     */
+    public onbarClickListener getOnBarClickListener() {
+        return onBarClickListener;
+    }
+
+
+    /**
+     * Attaches a onBarClickListener
+     */
+    public void setOnBarClickListener(onbarClickListener onBarClickListener) {
+        this.onBarClickListener = onBarClickListener;
+    }
+
+    public interface onbarClickListener {
```

Comment:
```suggestion
    public interface OnBarClickListener {
```
Make this change on every usage of the interface

Reply:
Done